omg/cli.py

```diff
@@ -1,94 +1,153 @@
-    p_mc.add_argument("--show-contents", dest="show_contents", action='store_true')
-    p_mc.set_defaults(func=machine_config)
-
-    # process args and call the corresponding function
-    args = parser.parse_args()
-    try:
-        func = args.func
-    except AttributeError:
-        parser.error("too few arguments")
-    try:
-        func(args)
-    except BrokenPipeError:
-        sys.stdout.close()
-    except KeyboardInterrupt:
-        pass
+CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])
+
+# Namespace related options shared by a few commands
+_global_namespace_options = [
+    click.option("--namespace", "-n", required=False, autocompletion=list_projects),
+    click.option("--all-namespaces", "-A", required=False, is_flag=True),
+]
+
+
+# Decorator lets us use namespace related options above
+def global_namespace_options(func):
+    for option in reversed(_global_namespace_options):
+        func = option(func)
+    return func
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+def cli():
+    pass
+
+
+@cli.command("use")
+@click.argument("mg_path", required=False,
+                type=click.Path(exists=True, file_okay=False, resolve_path=True, allow_dash=False))
+@click.option("--cwd", is_flag=True)
+def use_cmd(mg_path, cwd):
+    """
+    Select the must-gather to use
+    """
+    use(mg_path, cwd)
+
+
+@cli.command("project")
+@click.argument("name", required=False, autocompletion=list_projects)
+def project_cmd(name):
+    """
+    Display information about the current active project and existing projects
+    """
+    project(name)
+
+
+@cli.command("projects")
+def projects_cmd():
+    """
+    Display information about the current active project and existing projects
+    """
+    projects()
+
+
+@cli.command("get")
+@click.argument("objects", nargs=-1, autocompletion=complete_get)
+@click.option("--output", "-o", type=click.Choice(["yaml", "json", "wide"]))
+@global_namespace_options
+def get_cmd(objects, output, namespace, all_namespaces):
+    """
+    Display one or many resources
+    """
+    get_main(objects, output, namespace, all_namespaces)
+
+
+@cli.command("describe")
+@click.argument("object", nargs=-1)
+@global_namespace_options
+def describe_cmd(objects, namespace, all_namespaces):
+    """
+    This command joins many API calls together to form a detailed description of a given resource.
+    """
+    describe(None)
+
+
+@cli.command("logs")
+@click.argument("resource", autocompletion=list_pods)
+@click.option("--container", "-c", autocompletion=list_containers)
+@click.option("--previous", "-p", is_flag=True)
+@global_namespace_options  # TODO: Only support -n
+def logs_cmd(resource, container, previous, namespace, all_namespaces):
+    """
+    Display logs
+    """
+    log(resource, container, previous, namespace, False)
+
+
+@cli.command("whoami")
+def whoami_cmd():
+    """
+    Display who you are
+    """
+    whoami(None)
+
+
+@cli.command
+def version():
```

Comment:
@bostrt  `omg version` is not working. I think two minor fixes are required here.

First the the command name is missing. i.e, it should be `@cli.command("version")`

Second the function name `version` is conflicting as it is already imported  on top. I suggest renaming it to `version_cmd`